src/modules/mailer/domain/factories/email.factory.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { EmailFactory } from './email.factory';
+import { Email } from '../entities/email.entity';
+
+describe('EmailFactory', () => {
+  it('should create correct email with text and subject', () => {
+    const data = {
+      to: ['goodemail@gmail.com'],
+      subject: 'Good news',
+      text: 'The weather is good today!',
+    };
+    const email = EmailFactory.create(data);
+    expect(email).toBeDefined();
+    expect(email.text).toEqual(data.text);
+    expect(email.to).toEqual(data.to);
+    expect(email.subject).toEqual(data.subject);
+    expect(email.html).toBeUndefined();
+    expect(email).toBeInstanceOf(Email);
+  });
+
+  it('should create correct email with html and subject', () => {
+    const data = {
+      to: ['goodemail@gmail.com'],
+      subject: 'Good news',
+      html: '<b>The weather is good today!</b>',
+    };
+    const email = EmailFactory.create(data);
+    expect(email).toBeDefined();
+    expect(email.html).toEqual(data.html);
+    expect(email.to).toEqual(data.to);
+    expect(email.subject).toEqual(data.subject);
+    expect(email.text).toBeUndefined();
+    expect(email).toBeInstanceOf(Email);
+  });
+
+  it('should create correct email with several recipients', () => {
+    const data = {
+      to: [
+        'recipient1@gmail.com',
+        'recipient2@gmail.com',
+        'recipient3@gmail.com',
+        'recipient4@gmail.com',
+      ],
+      subject: 'Good news',
+      html: '<b>The weather is good today!</b>',
+    };
+    const email = EmailFactory.create(data);
+    expect(email).toBeDefined();
+    expect(email.html).toEqual(data.html);
+    expect(email.to).toEqual(data.to);
+    expect(email.subject).toEqual(data.subject);
+    expect(email.text).toBeUndefined();
+    expect(email).toBeInstanceOf(Email);
```

Comment:
How does this test specifically test several e-mails?)

Reply:
This line check it: expect(email.to).toEqual(data.to);